src/AdoNetCore.AseClient/AseParameter.cs

```diff
@@ -40,92 +42,6 @@ public AseParameter(string parameterName, object value) : this()
         {
             Value = value ?? throw new ArgumentNullException();
             ParameterName = parameterName;
-
-            if (value == DBNull.Value)
```

Comment:
Why's this getting removed? My guess would be that we're not supposed to infer type info? (noticed that the tests at the bottom have changed to expect length 0)

Reply:
The SAP version doesn't behave this way. I figured the likelihood that we get this inference right is pretty low. Do you depend on the AseDbType to set send the parameter over the wire, or do you infer it from Value.GetType()?

Maybe we should extract this logic into a helper class or something?

Reply:
Yeah, the over-the-wire `TdsDataType` is determined by the parameter's `DbType` / `AseDbType`.

Judging by the docs at https://docs.microsoft.com/en-us/dotnet/framework/data/adonet/configuring-parameters-and-parameter-data-types the over-the-wire type should be determined either from the supplied `DbType`, or the supplied `Value`

`TypeMap.cs` might be a suitable location to put it.

Reply:
I can't see any tests that fail because of this. Nevertheless I've added a `NetTypeToTdsMap` to the `TypeMap.cs` file and wired it into the GetTdsDataType method as a fallback if the DBType isn't set.

I haven't set the Size property - dunno if that's an issue also. Would be good to have a failing test...